examples/manipulation/behavior_cloning.py

```diff
@@ -0,0 +1,414 @@
+        checkpoint = torch.load(path, map_location=self._device, weights_only=False)
+        self._policy.load_state_dict(checkpoint["model_state_dict"])
+        self._optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+        self.current_iter = checkpoint["current_iter"]
+        print(f"Model loaded from {path}")
+
+    def load_finetuned_model(self, path: str) -> None:
+        """Load a fine-tuned model checkpoint."""
+        checkpoint = torch.load(path, map_location=self._device, weights_only=False)
+        self._policy.load_state_dict(checkpoint["model_state_dict"])
+        self._optimizer.load_state_dict(checkpoint["optimizer_state_dict"])
+        self._current_iter = checkpoint["current_iter"]
+        print(f"Fine-tuned model loaded from {path}")
+
+
+class ExperienceBuffer:
+    """A first-in-first-out buffer for experience replay."""
+
+    def __init__(
+        self,
+        num_envs: int,
+        max_size: int,
+        img_shape: tuple[int, int, int],
+        state_dim: int,
+        action_dim: int,
+        device: str = "cpu",
+    ):
+        self._num_envs = num_envs
+        self._max_size = max_size
+        self._img_shape = img_shape
+        self._state_dim = state_dim
+        self._action_dim = action_dim
+        self._device = device
+        self._ptr = 0
+        self._size = 0
+
+        # Buffers for data
+        self._rgb_obs = torch.empty(max_size, num_envs, *img_shape, dtype=torch.float32, device=device)
+        self._robot_pose = torch.empty(max_size, num_envs, state_dim, dtype=torch.float32, device=device)
+        self._object_poses = torch.empty(max_size, num_envs, 7, dtype=torch.float32, device=device)
+        self._actions = torch.empty(max_size, num_envs, action_dim, dtype=torch.float32, device=device)
+
+    def add(
+        self,
+        rgb_obs: torch.Tensor,
+        robot_pose: torch.Tensor,
+        object_poses: torch.Tensor,
+        actions: torch.Tensor,
+    ) -> None:
+        """Add experience to buffer."""
+        self._ptr = (self._ptr + 1) % self._max_size
+        self._rgb_obs[self._ptr] = rgb_obs
+        self._robot_pose[self._ptr] = robot_pose
+        self._object_poses[self._ptr] = object_poses
+        self._actions[self._ptr] = actions
+        self._size = min(self._size + 1, self._max_size)
+
+    def get_batches(self, num_mini_batches: int, num_epochs: int) -> Iterator[dict[str, torch.Tensor]]:
+        """Generate batches for training."""
+        # calculate the size of each mini-batch
+        batch_size = self._size // num_mini_batches
+        for _ in range(num_epochs):
+            indices = torch.randperm(self._size)
+            for batch_idx in range(0, self._size, batch_size):
+                batch_indices = indices[batch_idx : batch_idx + batch_size]
+
+                # Yield a mini-batch of data
+                yield {
+                    "rgb_obs": self._rgb_obs[batch_indices].view(-1, *self._img_shape),
```

Comment:
I would recommend to always use `reshape` everywhere, because `view` is just a useless legacy function that is redundant and I prefer to reduce the API coverage that I used from dependencies as a good practice to reduce maintenance burden. But it does clarify the intend for people that are not familiar with torch.

Reply:
switched to `reshape`